.github/workflows/swiftgen.yaml

```diff
@@ -0,0 +1,84 @@
+name: swiftgen
+
+on:
+  # Run on PRs and pushes to the default branch.
+  push:
+    branches: [main, stable]
+    paths:
+      - '.github/workflows/swiftgen.yaml'
+      - 'pkgs/swiftgen/**'
+  pull_request:
+    branches: [main, stable]
+    paths:
+      - '.github/workflows/swiftgen.yaml'
+      - 'pkgs/ffigen/**'
+      - 'pkgs/objective_c/**'
+      - 'pkgs/swift2objc/**'
+      - 'pkgs/swiftgen/**'
+  schedule:
+    - cron: "0 0 * * 0"
+
+env:
+  PUB_ENVIRONMENT: bot.github
+
+jobs:
+  # Check code formatting and static analysis.
+  analyze:
+    runs-on: macos-latest
+    defaults:
+      run:
+        working-directory: pkgs/swiftgen/
+    strategy:
+      fail-fast: false
+    steps:
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683
+      - uses: subosito/flutter-action@e938fdf56512cc96ef2f93601a5a40bde3801046
+        with:
+          channel: 'stable'
+      - id: install
+        name: Install dependencies
+        run: flutter pub get
+      - name: Check formatting
+        run: dart format --output=none --set-exit-if-changed .
+        if: always() && steps.install.outcome == 'success'
+      - name: Analyze code
+        run: dart analyze --fatal-infos
+        if: always() && steps.install.outcome == 'success'
+
+  test-mac:
+    needs: analyze
```

Comment:
why not merge into one bot?

Reply:
Until I finish migrating package:objective_c to native assets, some of these workflows need to run on Dart and some need to run on Flutter. I'll merge them once they can all run on Dart.